backend/src/resolvers/shout.js

```diff
@@ -0,0 +1,51 @@
+export default {
+  Mutation: {
+    shout: async (_object, params, context, _resolveInfo) => {
+      const { id, type } = params
+
+      const session = context.driver.session()
+      let transactionRes = await session.run(
+        `MATCH (node {id: $id})<-[:WROTE]-(userWritten:User), (user:User {id: $userId})
+          WHERE $type IN labels(node) AND NOT userWritten.id = $userId
```

Comment:
A future improvement could be that you can omit the `ShoutTypeEnum`. I mean: The server knows what you can shout and what not, right? So, instead of having the type of the shouted object sent to the server, you just send the id and the server checks if the labels of the shouted object intersects with a well-known set of shoutable objects (by a set of labels `['Project', 'Post', ...]`).

For now it is fine though!

Reply:
That is a concurrence between two things:
- KISS: Keep it stupid simple …
- But redundancy may be good for think and error checking: What type has the Webapp programmer thought to be shouted or followed and does this fit to what can be … on the server side?

I think you like to KISS … 😘

Reply:
😆